src/compiler/crystal/macros/methods.cr

```diff
@@ -1062,7 +1062,7 @@ module Crystal
 
     private def to_double_splat(trailing_string = "")
       MacroId.new(entries.join(", ") do |entry|
-        if Symbol.needs_quotes?(entry.key)
+        if Symbol.needs_quotes_for_names?(entry.key)
```

Comment:
Is the nodoc method used here because it avoids comparing things that will never match?

Reply:
What doesn't match?

The treatment here is the same; any operator or underscore key cannot be interpolated directly because such keys are always syntactically invalid. So this particular use changes the following

```crystal
{{ **{"+": 1, "-": -1} }} # => +: 1, -: -1
```

to `"+": 1, "-": -1`.

Reply:
Oooh... So that method is "needs quotes for named arguments"? If so, I think it should be named like that, even if it's a bit longer.

Reply:
And then this has nothing to do with symbols, right? Maybe the method should be in NamedArgument?

Reply:
`NamedTuple#to_s` depends on this method so it cannot be in the compiler source.

Reply:
Yeah, I was just thinking that... So this is fine! Thank you!